docs: add deprecation note next to IE 11 in browser-support page (#41147)
angular/angular: aio/content/guide/deprecations.md

- Clone the resulting value so that it has a new identity.
- Explicitly call [`ChangeDetectorRef.detectChanges()`](api/core/ChangeDetectorRef#detectchanges) to force the update.

### Internet Explorer 11

> Support for IE 11 has been deprecated and will be dropped in a future version. Supporting outdated browsers like this increases bundle size, code complexity, and test load, and also requires time and effort that could be spent on improvements to the framework. For example, fixing issues can be more difficult, as a straightforward fix for modern browsers could break old ones that have quirks due to not receiving updates from vendors.

Reviewer: ```suggestion
Angular support for Microsoft's Internet Explorer 11 (IE11) is deprecated and will end in a future version. Maintaining support for IE11 incurs ongoing costs, including increased bundle size, code complexity, and test load. Global usage of IE11 has fallen to a point where these costs no longer warrant the additional maintenance effort. Ending IE11 support additionally allows Angular to take advantage of platform APIs present only in evergreen browsers.
```

I will also want to come back and change "future version" here to the actual version once we decide on v13 or v14.

Reviewer 2: May I ask: Why is this information needed at all? IE11 is deprecated. Does the community really care about the fact that "Maintaining support for IE11 incurs ongoing costs?" Ongoing costs for whom? Google? The developer?

Can we just say "Microsoft has announced that they are deprecating IE11. Consequently, Angular is deprecating its support of this browser."

Also, we should not make statements like "in a future version." It's a problematic practice that can lead to out-dated documentation very quickly. If we must keep content like this paragraph, you can just say: "...is deprecated."

Reviewer: IMO providing the justification for deprecation is valuable because a small subset of developers are stuck supporting applications that _must_ work in IE11. Those developers will want to know why we're making this change, and pointing to Microsoft's own end-of-support for IE11 is good rationale.

Reviewer: Revisiting, the following statement is not accurate:
> Microsoft has announced that they are deprecating Internet Explorer 11. Consequently, Angular is deprecating its support of this browser.

Microsoft has not "deprecated" IE11. They have announced that they are removing IE11 support from their own flagship products and services. 

I still feel that elaborating on rationale is important. IMO only saying that our support for IE11 is deprecated without explaining _why_ would come off as opaque and aloof.

Author: I re-updated the deprecation statement. @aikidave and @jelbourn.

Reviewer 2: I won't block on this further, but I still question the language used here. 

@alan-agius4, can you please at least remove the "in a future version?" That clause is always problematic in the docs, and provides little to no additional information beyond what's already conveyed by saying something is deprecated.

"Angular support for Microsoft's Internet Explorer 11 (IE11) is deprecated." Should be sufficient.

Author: @aikidave, updated as per above suggestion.

> Support for IE 11 has been deprecated, this will be dropped in a future version. Supporting outdated browsers like these increases bundle size, code complexity, and test load, and also requires time and effort that could be spent on improvements to the framework. For example, fixing issues can be more difficult, as a straightforward fix for modern browsers could break old ones that have quirks due to not receiving updates from vendors.

Reviewer 3: ```suggestion
Support for IE 11 has been deprecated and will be dropped in a future version. Supporting outdated browsers like this increases bundle size, code complexity, and test load, and also requires time and effort that could be spent on improvements to the framework. For example, fixing issues can be more difficult, as a straightforward fix for modern browsers could break old ones that have quirks due to not receiving updates from vendors.
```

> Microsoft announched that it's 365 apps and services will no longer support Internet Explorer 11 (IE 11) from August 17, 2020. Additionally, Microsoft dropped support for IE 11 from Microsoft Teams web app in November 30, 2020. 

Reviewer 3: ```suggestion
Microsoft announced that its 365 apps and services no longer support Internet Explorer 11 (IE 11) from August 17, 2020. Additionally, Microsoft dropped support for IE 11 from Microsoft Teams web app in November 30, 2020. 
```

> Microsoft announced that its 365 apps and services no longer support Internet Explorer 11 (IE 11) from August 17, 2020. Additionally, Microsoft dropped support for IE 11 from Microsoft Teams web app in November 30, 2020. 
> 
> For more information, see [Microsoft 365 apps say farewell to Internet Explorer 11](https://techcommunity.microsoft.com/t5/microsoft-365-blog/microsoft-365-apps-say-farewell-to-internet-explorer-11-and/ba-p/1591666).

Reviewer: ```suggestion
Microsoft announced that its 365 services will no longer support IE11 starting August 17, 2021. Microsoft previously ended support for IE11 in Microsoft Teams on November 30, 2020. For more information, see [Microsoft 365 apps say farewell to Internet Explorer 11](https://techcommunity.microsoft.com/t5/microsoft-365-blog/microsoft-365-apps-say-farewell-to-internet-explorer-11-and/ba-p/1591666).
```

some syntax changes and combining these two paragraphs into one

Reviewer 2: Again, may I ask why this information is important to Angular developers? Are we just trying to justify why we're deprecating our own support for IE 11?
